source/extensions/common/workload_discovery/api.cc

```diff
@@ -46,10 +46,22 @@ Istio::Common::WorkloadMetadataObject convert(const istio::workload::Workload& w
   default:
     break;
   }
+  std::string ns = workload.namespace_();
+  std::string trust_domain = workload.trust_domain();
+  // Trust domain may be elided if it's equal to "cluster.local"
+  if (trust_domain.empty()) {
+    trust_domain = "cluster.local";
```

Comment:
cluster.local is strongly discouraged - I think it should be explicit.

Reply:
This is also from the proto. In my tests the trust domain was elided as well